### PR TITLE
Add 'file' support to parameters

### PR DESCRIPTION
**Pull request by analogue**
Useful documentation for the Requests lib on file support - http://docs.python-requests.org/en/latest/user/advanced/#post-multiple-multipart-encoded-files

### CODE REVIEW

File: bravado/mapping/operation.py [hunk 1]
```diff
@@ -63,7 +75,7 @@ def build_params(self):
         param_specs = op_param_specs + path_param_specs
 
         for param_spec in param_specs:
-            param = Param(self.swagger_spec, param_spec)
+            param = Param(self.swagger_spec, self, param_spec)
```
Comment (prat0318) [inline bravado/mapping/operation.py]: is it totally necessary to send the op in the request? how about just the `consumes` field?

Comment (analogue) [inline bravado/mapping/operation.py]: `operation_id` is also used. I predict `produces` will also be used at some point too.

File: bravado/mapping/param.py [hunk 2]
```diff
@@ -108,3 +112,44 @@ def marshal_param(param, value, request):
...
+    if request.get('file') is None:
+        # support multiple files by default by setting to an empty array
+        request['file'] = []
+
+        # The http client should take care of setting the content-type header
+        # to 'multipart/form-data'. Just verify that the swagger spec is
+        # conformant
+        expected_mime_type = 'multipart/form-data'
+        if expected_mime_type not in param.op.consumes:
+            raise SwaggerMappingError((
```
Comment (prat0318) [inline bravado/mapping/param.py]: This should ideally be the job of rich validations of `swagger_spec_validator`.

Comment (prat0318) [inline bravado/mapping/param.py]: oh my bad. `consumes` is a part of the `request` and cant be checked with ssv.

Comment (dnephin) [inline bravado/mapping/param.py]: I think you're right, this should be part of the static validation of the schema, performed by ssv.

File: bravado/mapping/param.py [hunk 3]
```diff
@@ -108,3 +112,44 @@ def marshal_param(param, value, request):
...
+def add_formdata(param, value, request):
+    """Add a parameter as formdata to the given request.
+
+    :type param: :class;`bravado.mapping.param.Param`
+    :param value: parameter value
+    :type request: dict
+    """
+    if request.get('data') is None:
+        request['data'] = {}
+    request['data'][param.name] = value
```
Comment (dnephin) [inline bravado/mapping/param.py]: Minor: you can do this as:

```python
request.setdetault('data', {})[param.name] = value
```

File: bravado/mapping/param.py (hunk 2 above)
Comment (analogue) [inline bravado/mapping/param.py]: Add TODO and ticketed ssv

File: bravado/mapping/param.py (hunk 3 above)
Comment (analogue) [inline bravado/mapping/param.py]: Fixed

**Pull request merged**